optica.rb

```diff
@@ -68,15 +68,28 @@ class Optica < Sinatra::Base
       halt(403) unless ip == header.split(',').first
     end
 
+    # update main store
     begin
       merged_data = settings.store.add(ip, data)
-      settings.events.send(merged_data.merge("event" => data))
     rescue
       halt(500)
     end
 
+    # publish update event
+    error_msg = ""
```

Comment:
There's no need to rescue from `Exception` here. It may catch undesirables like `SIGTERM`. A simple `rescue => e` is sufficient. Your common case can be simplified to `e = nil` and therefore —

```
message += " -- [warning] failed to publish to rabbitmq: #{e.to_s}" if !e.nil?
```

Or, even better, just append to the message inside the rescue block.